### PR TITLE
Upload File Types

### LINKED ISSUE #2494: Upload File Types
https://github.com/ProcessMaker/screen-builder/issues/421

### PR DESCRIPTION
**Pull request by marcoAntonioNina**
closes #2494 

## Changes
- add property "File accepted" and validation by file.

## Test

- create form with upload file
- enter files accepted
- create process with form
- Select files not allowed and allowed.

## Screenshot

configuration
image

file not allowed
image

file allowed
image

### CODE REVIEW

File: resources/js/processes/screen-builder/components/form/file-upload.vue [hunk 1]
```diff
@@ -93,10 +94,27 @@ export default {
...
       },
+      attrs: {
+        accept: this.type
+      },
       reRenderKey: 0,
     };
   },
   methods: {
+    addFile(file) {
+      if (this.type) {
```
Comment (velkymx) [inline resources/js/processes/screen-builder/components/form/file-upload.vue]: Why are we storing the type in a string to begin with? Wouldn't it be cleaner if we store the type as an object from the beginning and just use the `type.includes('ext')` method instead?

File: resources/js/processes/screen-builder/components/form/file-upload.vue [hunk 2]
```diff
@@ -36,13 +37,13 @@ const uniqIdsMixin = createUniqIdsMixin();
 export default {
   components: uploader,
   mixins: [uniqIdsMixin],
-  props: ["label", "error", "helper", "name", "value", "controlClass", "endpoint"],
+  props: ["label", "error", "helper", "name", "value", "controlClass", "endpoint", "type"],
```
Comment (velkymx) [inline resources/js/processes/screen-builder/components/form/file-upload.vue]: We should use a better name then type - `accept` is the correct term

File: resources/js/processes/screen-builder/components/form/file-upload.vue [hunk 3]
```diff
@@ -10,12 +10,13 @@
...
       @file-success="fileUploaded"
+      @file-added="addFile"
     >
       <uploader-unsupport></uploader-unsupport>
 
       <uploader-drop id="uploaderMain" class="form-control-file">
-        <p>{{$t('Drop a file here to upload or')}}</p>
-        <uploader-btn id="submitFile" class="btn btn-secondary text-white">{{$t('select file')}}</uploader-btn>
+        <p>{{ $t('Drop a file here to upload or') }}</p>
+        <uploader-btn id="submitFile" class="btn btn-secondary text-white" :attrs="attrs">{{ $t('select file') }}</uploader-btn>
```
Comment (velkymx) [inline resources/js/processes/screen-builder/components/form/file-upload.vue]: attrs is wrong, use `accept`

**Comment by marcoAntonioNina** [comment]
@velkymx fix comments.

**Pull request merged**